src/components/engine/contract-subscription/add-contract-subs-button.tsx

```diff
@@ -0,0 +1,178 @@
+import {
+  CreateEngineContractSubscription,
+  useEngineAddContractSubscription,
+} from "@3rdweb-sdk/react/hooks/useEngine";
+import {
+  Flex,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  useDisclosure,
+  Icon,
+  FormControl,
+  Input,
+  UseDisclosureReturn,
+} from "@chakra-ui/react";
+import { useTrack } from "hooks/analytics/useTrack";
+import { useTxNotifications } from "hooks/useTxNotifications";
+import { useForm } from "react-hook-form";
+import { Button, FormLabel } from "tw-components";
+import { AiOutlinePlusCircle } from "react-icons/ai";
+import { useAllChainsData } from "hooks/chains/allChains";
+import { NetworkDropdown } from "components/contract-components/contract-publish-form/NetworkDropdown";
+import { isAddress } from "ethers/lib/utils";
+
+interface AddContractSubscriptionButtonProps {
+  instanceUrl: string;
+}
+
+export const AddContractSubscriptionButton: React.FC<
+  AddContractSubscriptionButtonProps
+> = ({ instanceUrl }) => {
+  const disclosure = useDisclosure();
+
+  return (
+    <>
+      <Button
+        onClick={disclosure.onOpen}
+        variant="ghost"
+        size="sm"
+        leftIcon={<Icon as={AiOutlinePlusCircle} boxSize={6} />}
+        colorScheme="primary"
+        w="fit-content"
+      >
+        Add Contract Subscription
+      </Button>
+
+      {disclosure.isOpen && (
+        <AddModal instanceUrl={instanceUrl} disclosure={disclosure} />
+      )}
+    </>
+  );
+};
+
+export interface AddModalInput {
+  chainId: string;
+  contractAddress: string;
+}
+
+const AddModal = ({
+  instanceUrl,
+  disclosure,
+}: {
+  instanceUrl: string;
+  disclosure: UseDisclosureReturn;
+}) => {
+  const { mutate: addContractSubscription } =
+    useEngineAddContractSubscription(instanceUrl);
+  const { chainIdToChainRecord } = useAllChainsData();
+  const trackEvent = useTrack();
+  const { onSuccess, onError } = useTxNotifications(
+    "Contract Subscription created successfully.",
+    "Failed to create contract subscription.",
+  );
+
+  const form = useForm<AddModalInput>({
+    defaultValues: {
+      chainId: "84532",
+    },
```

Comment:
do we mean to have a default value for this?

Reply:
I thought having a default would be good, to make it easier for user to understand the input req.